src/search.c

```diff
@@ -499,6 +578,26 @@ static void do_search(struct manifest *MoM, char search_type, char *search_term)
 			continue;
 		}
 
+		/* record contentsize and includes for install size calculation */
+		struct bundle_result *bundle = NULL;
+		bundle = calloc(sizeof(struct bundle_result), 1);
+		/* copy relevant information over for future use */
+		strncpy(bundle->bundle_name, subman->component, BUNDLE_NAME_MAXLEN - 1);
+		bundle->topsize = subman->contentsize;
+		/* do a deep copy of the includes list
+		 * first list_clone so we alloc enough room */
+		bundle->includes = list_clone(subman->includes);
+		/* now strdup all the includes into place */
+		struct list *sub_incs = list_head(subman->includes);
+		struct list *bundle_incs = list_head(bundle->includes);
+		while (sub_incs && bundle_incs) {
```

Comment:
Probably a waste of time, but I would prefer this to be a for loop.

Not clear why you are stopping on either sub_incs or bundle_incs, you cloned them so they had better be the same length.

Why do we need to strdup the filenames? Is it just to save the effort of not setting the names to NULL when cleaning up?

Reply:
> Why do we need to strdup the filenames?

If I don't strdup the includes list I lose the pointers when prepending to the list on the next iteration.